src/Simulation/Simulators/QuantumSimulator/ApplyControlledX.cs

```diff
@@ -2,12 +2,13 @@
 // Licensed under the MIT License.
 
 using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Intrinsic.Interfaces;
 
 namespace Microsoft.Quantum.Simulation.Simulators
 {
     public partial class QuantumSimulator
     {
-        public virtual void ApplyControlledX__Body(Qubit control, Qubit target)
```

Comment:
What is the reason for the change in behavior of these from `public virtual` to the implicit `private`?

Reply:
This is actually an interesting thing that falls out of using interface methods that have the same name. To differentiate `Body` offered by `IIntrinsicApplyControlledX` from `Body` offered by `IIntrinsicApplyControlledZ`, we need to fully qualify the name by saying which interface it comes from. That's why the new signature becomes `void IIntrinsicApplyControlledX.Body`. If I add `public` to this signature, it gives me a compiler error:
```
The modifier 'public' is not valid for this item [Microsoft.Quantum.Simulators] csharp(CS0106)
```
That is because now that the method is fully qualified, it is known to come from a public interface, which means it is public by definition. Including any access modifier, even one that agrees with the defined one, is considered an error.